integration_test/test/recording_endpoint_test.exs

```diff
@@ -129,12 +144,44 @@ defmodule Membrane.RTC.RecordingEndpointTest do
     validate_report(report_file_path)
   end
 
+  setup :verify_on_exit!
+  setup :set_mox_from_context
+
+  test "recording endpoint with aws storage", %{rtc_engine: rtc_engine} do
+    recording_endpoint_id = "recording-endpoint"
+    video_file_endpoint_id = "video-file-endpoint"
+
+    video_file_path = Path.join(@fixtures_dir, "recorded_video.h264")
+    setup_mock_http_request()
+
+    recording_endpoint = %{
+      create_recording_endpoint(rtc_engine, @path)
+      | stores: [{Storage.S3, %{credentials: @credentials}}]
+    }
+
+    video_file_endpoint = create_video_file_endpoint(rtc_engine, video_file_path)
+
+    :ok = Engine.add_endpoint(rtc_engine, recording_endpoint, id: recording_endpoint_id)
+    :ok = Engine.add_endpoint(rtc_engine, video_file_endpoint, id: video_file_endpoint_id)
+
+    assert_receive %TrackAdded{endpoint_id: ^video_file_endpoint_id}, @tracks_added_delay
+    assert_receive %TrackRemoved{endpoint_id: ^video_file_endpoint_id}, @tracks_removed_delay
+
+    Engine.remove_endpoint(rtc_engine, recording_endpoint_id)
+    assert_receive %EndpointRemoved{endpoint_id: ^recording_endpoint_id}
+
+    assert_receive :upload_initialized
+    assert_receive :chunk_uploaded
+    assert_receive :upload_completed
+    assert_receive :report_uploaded
+  end
+
   defp create_recording_endpoint(rtc_engine, output_dir) do
     %RecordingEndpoint{
       rtc_engine: rtc_engine,
       recording_id: "id",
       output_dir: output_dir,
-      stores: [Storage.File]
+      stores: [{Storage.File, nil}]
```

Comment:
I don't understand this idea to be honest

Reply:
Because if you would have a parameter, you could simply pass the value instead of doing this:
https://github.com/jellyfish-dev/membrane_rtc_engine/blob/8cf6e5d8cce52de23c0cf59c31af6de799ab7942/integration_test/test/recording_endpoint_test.exs#L157-L160